src/main/java/mate/academy/springboot/datajpa/service/CategoryServiceImpl.java

```diff
@@ -0,0 +1,44 @@
+package mate.academy.springboot.datajpa.service;
+
+import java.util.NoSuchElementException;
+import mate.academy.springboot.datajpa.model.Category;
+import mate.academy.springboot.datajpa.repository.CategoryRepository;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CategoryServiceImpl implements CategoryService {
+    private final CategoryRepository categoryRepository;
+
+    public CategoryServiceImpl(CategoryRepository categoryRepository) {
+        this.categoryRepository = categoryRepository;
+    }
+
+    @Override
+    public Category create(Category category) {
+        return categoryRepository.save(category);
+    }
+
+    @Override
+    public Category get(Long id) {
+        return categoryRepository.findById(id).orElseThrow(() ->
+                new NoSuchElementException("couldn't get category by id: " + id));
+    }
+
+    @Override
+    public void delete(Long id) {
+        if (categoryRepository.existsById(id)) {
+            categoryRepository.deleteById(id);
+        } else {
+            throw new NoSuchElementException("couldn't delete category by id: " + id);
+        }
```

Comment:
This approach "fails fast" by handling the exception first, reducing the need for nested code. Since an exception ends the method, there's no need for an else clause, making the code cleaner. Let's do it this way everywhere
```suggestion
        if (!categoryRepository.existsById(id)) {
            categoryRepository.deleteById(id);
        } 
        throw new NoSuchElementException("Couldn't delete category by id: " + id);
```